### PR TITLE
Add calendar view

### LINKED ISSUE #1: Add basic calendar view
Have to add 2 pages and a header, which looks like this:
image

### PR DESCRIPTION
**Pull request by f0xer1**
Deployed basic calendar view: https://stupendous-souffle-cd7e8d.netlify.app/
Added two calendar pages: month and year with the current day, where we can switch to other years and months and press the button to return to the current. Also added adaptability.

Basic view:
image

Adaptive:
image

### CODE REVIEW

File: src/app/models/CalendarItem.ts
Comment (AlexanderAndreyev) [inline src/app/models/CalendarItem.ts:1]: name should like `calendar-item.model.ts`. Pascal case isn't used a lot in Angular apps

File: src/app/modules/calendar/calendar-month/calendar-month.component.css [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+}
+
+button[mat-raised-button] {
+  --mdc-protected-button-container-shape: 5px;
+  --mdc-protected-button-container-elevation-shadow: none;
+  margin-left: 16px;
+  color: gray;
+  background-color: #f1f0f0;
+  font-weight: 500;
+}
```
Comment (AlexanderAndreyev) [inline src/app/modules/calendar/calendar-month/calendar-month.component.css]: please remove customizations - we don't need them :)

File: src/app/modules/calendar/calendar-month/calendar-month.component.css [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+
+.dropdown select:hover {
+  border-color: #999;
+}
+
+.dropdown select:focus {
+  border-color: black;
+  box-shadow: 0 0 0 2px #f1f0f0;
+}
```
Comment (AlexanderAndreyev) [inline src/app/modules/calendar/calendar-month/calendar-month.component.css]: not really good approach to style by tag

File: src/app/modules/calendar/calendar-month/calendar-month.component.html [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    </div>
+  </div>
+  <div class="calendar-header">
+    <div class="calendar-day-name">Monday</div>
+    <div class="calendar-day-name">Tuesday</div>
+    <div class="calendar-day-name">Wednesday</div>
+    <div class="calendar-day-name">Thursday</div>
+    <div class="calendar-day-name">Friday</div>
+    <div class="calendar-day-name">Saturday</div>
+    <div class="calendar-day-name">Sunday</div>
```
Comment (AlexanderAndreyev) [inline src/app/modules/calendar/calendar-month/calendar-month.component.html]: mb some ngFor?

File: src/app/modules/calendar/calendar-month/calendar-month.component.html [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+        <mat-icon>chevron_right</mat-icon>
+      </button>
+      <button mat-raised-button (click)="currentMonth()">Today</button>
+    </mat-toolbar-row>
+    <div class="dropdown">
+      <select>
+        <option value="">Filter by absence type</option>
+        <option value="1">Sick Leave</option>
+        <option value="2">Vacation</option>
+      </select>
```
Comment (AlexanderAndreyev) [inline src/app/modules/calendar/calendar-month/calendar-month.component.html]: please use Angular Material select

File: src/app/modules/calendar/calendar-month/calendar-month.component.html [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+    <div class="calendar-day-name">Tuesday</div>
+    <div class="calendar-day-name">Wednesday</div>
+    <div class="calendar-day-name">Thursday</div>
+    <div class="calendar-day-name">Friday</div>
+    <div class="calendar-day-name">Saturday</div>
+    <div class="calendar-day-name">Sunday</div>
+  </div>
+  <div class="calendar-body">
+    <div class="calendar-row" *ngFor="let row of calendar ">
+      <div class="calendar-day" *ngFor="let c of row"
```
Comment (AlexanderAndreyev) [inline src/app/modules/calendar/calendar-month/calendar-month.component.html]: please change to readable name from `c`

File: src/app/modules/calendar/calendar-month/calendar-month.component.ts [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+import {NgClass, NgForOf, NgIf} from "@angular/common";
+import {CalendarService} from "../../../services/CalendarService";
+import moment from "moment";
+import {CalendarItem} from "../../../models/CalendarItem";
+import {MatFormFieldModule} from "@angular/material/form-field";
+import {MatInputModule} from "@angular/material/input";
+import {MatSelectModule} from "@angular/material/select";
+import {MatToolbar, MatToolbarRow} from "@angular/material/toolbar";
+import {MatIcon} from "@angular/material/icon";
+import {MatButton, MatIconButton} from "@angular/material/button";
```
Comment (AlexanderAndreyev) [inline src/app/modules/calendar/calendar-month/calendar-month.component.ts]: would be awesome to use some pattern for imports

for example:
1) 3rd party libs imports (Angular, material etc)
2) internal imports

please check how to setup eslint for automatization such a case and add a plugin that can help with imports sorting

File: src/app/modules/calendar/calendar-month/calendar-month.component.ts [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+    NgForOf,
+    NgClass,
+    MatFormFieldModule, MatInputModule, MatSelectModule, MatToolbarRow, MatToolbar, MatIcon, MatIconButton, MatButton
+  ],
+  providers: [CalendarService],
+  templateUrl: './calendar-month.component.html',
+  styleUrl: './calendar-month.component.css'
+})
+export class CalendarMonthComponent implements OnInit {
+  date: moment.Moment = moment();
```
Comment (AlexanderAndreyev) [inline src/app/modules/calendar/calendar-month/calendar-month.component.ts]: what date exactly?

File: src/app/modules/calendar/calendar-month/calendar-month.component.ts [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+  calendar: Array<CalendarItem[]> = [];
```
Comment (AlexanderAndreyev) [inline src/app/modules/calendar/calendar-month/calendar-month.component.ts]: Array<CalendarItem[]> -> CalendarItem[][]

File: src/app/modules/calendar/calendar-year/calendar-year.component.css [hunk 9]
```diff
@@ -0,0 +1,157 @@
+.calendar {
+  display: flex;
+  flex-direction: column;
+  width: 100%;
+  max-width: 1200px;
+  margin: 0 auto;
```
Comment (AlexanderAndreyev) [inline src/app/modules/calendar/calendar-year/calendar-year.component.css]: I saw this style already. Mb better to move to the global css file to reuse?

File: src/app/modules/calendar/calendar-year/calendar-year.component.css
Comment (AlexanderAndreyev) [inline src/app/modules/calendar/calendar-year/calendar-year.component.css:1]: file looks familiar :)

File: src/app/modules/calendar/calendar-year/calendar-year.component.html [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+        <mat-icon>chevron_right</mat-icon>
+      </button>
+      <button mat-raised-button (click)="currentYear()">Current</button>
+    </mat-toolbar-row>
+    <div class="dropdown">
+      <select>
+        <option value="">Filter by absence type</option>
+        <option value="1">Sick Leave</option>
+        <option value="2">Vacation</option>
+      </select>
```
Comment (AlexanderAndreyev) [inline src/app/modules/calendar/calendar-year/calendar-year.component.html]: material please

File: src/app/modules/calendar/calendar-year/calendar-year.component.html [hunk 11]
```diff
@@ -0,0 +1,35 @@
...
+    </div>
+  </div>
+  <div class="calendar-box">
+    <div class="calendar-month" *ngFor="let month of calendar; let monthIndex = index">
+      <div class="calendar-month-name">
+        {{ monthArr[monthIndex] }}
+      </div>
+      <div class="calendar-row" *ngFor="let row of month">
+        <div class="calendar-day" *ngFor="let c of row">
```
Comment (AlexanderAndreyev) [inline src/app/modules/calendar/calendar-year/calendar-year.component.html]: I saw it)

File: src/app/modules/header/header.component.css [hunk 12]
```diff
@@ -0,0 +1,33 @@
...
+  align-items: center;
+  max-width: 1200px;
+  width: 100%;
+  margin: 0 auto;
+  border-bottom: #eaeaea 1px solid;
+  padding-top: 10px;
+}
+
+
+.logo-container h3 {
```
Comment (AlexanderAndreyev) [inline src/app/modules/header/header.component.css]: tag styling - 🔴

File: src/app/services/CalendarService.ts [hunk 13]
```diff
@@ -0,0 +1,58 @@
...
+  public createCalendar(month: moment.Moment): Array<CalendarItem[]> {
+    const daysInMonth = month.daysInMonth();
+    const daysBefore = this.weekdaysShort.indexOf(month.startOf('months').format('ddd'));
+    const daysAfter = this.weekdaysShort.length - 1 - this.weekdaysShort
+      .indexOf(month.endOf('months').format('ddd'));
+
+    const calendar: CalendarItem[] = [];
+    const clone = month.startOf('months').clone().subtract(daysBefore, 'days');
+
+    for (let i = 0; i < daysBefore; i++) {
```
Comment (AlexanderAndreyev) [inline src/app/services/CalendarService.ts]: `for of` mb?

**Review by AlexanderAndreyev — CHANGES_REQUESTED** [review]

File: src/app/services/CalendarService.ts (hunk 13 above)
Comment (f0xer1) [reply src/app/services/CalendarService.ts]: daysBefore, daysAfter, and daysInMonth are single numbers, not arrays. I didn't delete the for i loop; instead, I reduced the three loops into one using a ternary operator

File: src/app/models/calendar-item.model.ts [hunk 14]
```diff
@@ -1,4 +1,4 @@
-export interface CalendarItem {
+export interface CalendarItemModel {
```
Comment (OleksandrAndrieiev-KSO) [inline src/app/models/calendar-item.model.ts:1]: you can skip `Model` suffix in naming itself :)

File: src/app/modules/calendar/calendar-month/calendar-month.component.ts [hunk 15]
```diff
@@ -24,28 +24,29 @@ import {MatButton, MatIconButton} from "@angular/material/button";
   styleUrl: './calendar-month.component.css'
 })
 export class CalendarMonthComponent implements OnInit {
-  date: moment.Moment = moment();
-  calendar: Array<CalendarItem[]> = [];
+  currentData: moment.Moment = moment();
```
Comment (OleksandrAndrieiev-KSO) [inline src/app/modules/calendar/calendar-month/calendar-month.component.ts:28]: currentData -> currentDate

File: src/app/modules/calendar/calendar-year/calendar-year.component.ts [hunk 16]
```diff
@@ -31,29 +31,29 @@ import {MatToolbarRow} from "@angular/material/toolbar";
   styleUrl: './calendar-year.component.css'
 })
 export class CalendarYearComponent implements OnInit {
-  date: moment.Moment = moment();
-  calendar: Array<CalendarItem[][]> = [];
-  monthArr: String[] = moment.months();
+  currentData: moment.Moment = moment();
```
Comment (OleksandrAndrieiev-KSO) [inline src/app/modules/calendar/calendar-year/calendar-year.component.ts:35]: same :)

File: src/styles.css [hunk 17]
```diff
@@ -2,11 +2,78 @@
...
+html, body {
+  height: 100%;
+}
+
+body {
+  margin: 0;
+  font-family: Roboto, "Helvetica Neue", sans-serif;
+}
+
+
```
Comment (OleksandrAndrieiev-KSO) [inline src/styles.css]: extra space

File: src/styles.css [hunk 18]
```diff
@@ -2,11 +2,78 @@
...
+  width: 100%;
+}
+.calendar-month-name, .calendar-day-name {
+  text-align: center;
+  background-color: #f1f0f0;
+  color: #454444;
+  padding: 10px;
+  box-sizing: border-box;
+  width: 100%;
+}
```
Comment (OleksandrAndrieiev-KSO) [inline src/styles.css:84]: please space between the rules

**Review by OleksandrAndrieiev-KSO — CHANGES_REQUESTED** [review]

File: src/app/models/calendar-item.model.ts [hunk 19]
```diff
@@ -1,4 +1,4 @@
-export interface CalendarItem {
```
Comment (AlexanderAndreyev) [inline src/app/models/calendar-item.model.ts]: you can remove `Model` suffix in the name :)

File: src/app/modules/calendar/calendar-month/calendar-month.component.ts (hunk 15 above)
Comment (AlexanderAndreyev) [inline src/app/modules/calendar/calendar-month/calendar-month.component.ts]: currentData -> currentDate

File: src/app/modules/calendar/calendar-year/calendar-year.component.ts (hunk 16 above)
Comment (AlexanderAndreyev) [inline src/app/modules/calendar/calendar-year/calendar-year.component.ts]: same

File: src/app/modules/calendar/calendar-year/calendar-year.component.ts [hunk 20]
```diff
@@ -31,29 +31,29 @@ import {MatToolbarRow} from "@angular/material/toolbar";
...
+  calendar: CalendarItemModel[][][] = [];
+  monthArr: string[] = moment.months();
```
Comment (AlexanderAndreyev) [inline src/app/modules/calendar/calendar-year/calendar-year.component.ts]: monthArr -> monthList ?

File: src/styles.css [hunk 21]
```diff
@@ -2,11 +2,78 @@
...
+  flex-wrap: wrap;
+  border-radius: 20px;
+  padding: 10px;
+}
+
+.nav-month-box {
+  display: flex;
+  justify-content: center;
+  align-items: center;
+}
```
Comment (AlexanderAndreyev) [inline src/styles.css:38]: please add spaces between the rules

File: src/styles.css [hunk 22]
```diff
@@ -2,11 +2,78 @@
...
-body { margin: 0; font-family: Roboto, "Helvetica Neue", sans-serif; }
+html, body {
+  height: 100%;
+}
+
+body {
+  margin: 0;
+  font-family: Roboto, "Helvetica Neue", sans-serif;
+}
+
```
Comment (AlexanderAndreyev) [inline src/styles.css:23]: remove extra space please

**Review by AlexanderAndreyev — CHANGES_REQUESTED** [review]

**Review by AlexanderAndreyev — APPROVED** [review]

**Pull request merged**